### PR TITLE
docs: change wording in ComponentDecorator to incorporate components without NgModule

### PR DESCRIPTION
**Pull request by ChinoUkaegbu**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
The documentation insists that a component must belong to an NgModule in order for it to be available to another component or application.

Issue Number: #57689 

## What is the new behavior?
The documentation still provides instructions to make a component a member of an NgModule, but only when running Angular with NgModule

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: packages/core/src/metadata/directives.ts [hunk 1]
```diff
@@ -401,9 +401,9 @@ export interface ComponentDecorator {
    * Unlike other directives, only one component can be instantiated for a given element in a
    * template.
    *
-   * A component must belong to an NgModule in order for it to be available
-   * to another component or application. To make it a member of an NgModule,
-   * list it in the `declarations` field of the `NgModule` metadata.
+   * When running Angular with NgModule, a component must belong to an NgModule in
```
Comment (JeanMeche) [inline packages/core/src/metadata/directives.ts:404]: I'm wondering if we shouldn't drop that section altogether as we're moving toward standalone by default.

Comment (ChinoUkaegbu) [reply packages/core/src/metadata/directives.ts:404]: If using NgModule is still going to be an option for users, then it's worth keeping imo! Or perhaps changing 'When running' to 'If running' so it seems more optional?

Comment (JeanMeche) [reply packages/core/src/metadata/directives.ts:404]: The jsdoc only mentionning NgModules and not standalone components is an issue to me. The docs should clearly indicate the recommendation toward standalone.

Comment (ChinoUkaegbu) [reply packages/core/src/metadata/directives.ts:404]: Would adding "However, Angular now recommends using standalone components, which do not require an NgModule and can be directly imported and used in other components or applications." or something to that effect resolve this or is that too wordy and would it be better to just remove that section entirely?

Comment (JeanMeche) [reply packages/core/src/metadata/directives.ts:404]: Yeah that looks good !

Comment (ChinoUkaegbu) [reply packages/core/src/metadata/directives.ts:404]: Made the change!

Comment (ChinoUkaegbu) [reply packages/core/src/metadata/directives.ts]: Starting with the recommended method makes sense to me, I've edited it

File: packages/core/src/metadata/directives.ts [hunk 2]
```diff
@@ -401,9 +401,11 @@ export interface ComponentDecorator {
...
    * template.
    *
-   * A component must belong to an NgModule in order for it to be available
-   * to another component or application. To make it a member of an NgModule,
-   * list it in the `declarations` field of the `NgModule` metadata.
+   * Angular now recommends using standalone components, which do not require an 
+   * NgModule and can be directly imported and used in other components or applications.
+   * However, if running Angular with NgModule, a component must belong to an NgModule in
+   * order for it to be available to another component or application. To make it a
+   * member of an NgModule, list it in the `declarations` field of the `NgModule` metadata.
```
Comment (JeanMeche) [inline packages/core/src/metadata/directives.ts:408]: We shouldn't use any time relative statement, 

Wdyt of something like that ? 

```suggestion
   * Standalone components can be directly imported in any other standalone component or NgModule.
   * NgModule based apps on the other hand require components to belong to an NgModule in
   * order for it to be available to another component or application. To make it a
   * member of an NgModule, list it in the `declarations` field of the `NgModule` metadata.
```

**Review by JeanMeche — APPROVED** [review]

**Comment by JeanMeche** [comment]
Could you please squash all the commit into one, we'd like this change to be a single commit. Thank you.
